django/db/backends/postgresql/client.py

```diff
@@ -54,8 +55,11 @@ def runshell_db(cls, conn_params):
                     # If the current locale can't encode the data, we let
                     # the user input the password manually.
                     pass
+            default_handler = signal.getsignal(signal.SIGINT)
+            signal.signal(signal.SIGINT, lambda sig, frame: None)
             subprocess.check_call(args)
         finally:
+            signal.signal(signal.SIGINT, default_handler)
```

Comment:
If we hit an exception in the block before the default_handler is recorded, then I think we will hit an `UnboundLocalError` here because `default_handler` is not assigned. This is obviously pretty unlikely, since there aren't many exceptions that can be thrown from the previous block. However, it would be slightly safer to have a narrower `try ... finally` block just around the signal-handling stuff.